lib/auth/permissions.go

```diff
@@ -727,12 +723,53 @@ type BuiltinRoleSet struct {
 	services.RoleSet
 }
 
+func (set BuiltinRoleSet) CheckAccessToApp(login string, app *services.App, mfa services.AccessMFAParams) error {
```

Comment:
@klizhentas I suspect that a lot of basic functionality could break if a proxy can't read the list of ssh/k8s/db/webapp instances

The idea was to tighten this part with role-based exception cases in a followup PR. No actual release would include this blanked exception as is.

Reply:
* connect access - fails because builtin roles don't have `logins`, `kube_users`, `db_users`, etc
* write access - subject to RBAC write verbs as before; labels didn't apply there in the first place
* read/list access - subject to RBAC read verbs as before; but you can now read SSH nodes as any service (not just proxy, due to custom logic in `filterNodes`)